app/src/main/java/com/zulip/android/networking/AsyncEdit.java

```diff
@@ -0,0 +1,45 @@
+package com.zulip.android.networking;
+
+import com.zulip.android.R;
+import com.zulip.android.ZulipApp;
+import com.zulip.android.activities.ZulipActivity;
+import com.zulip.android.models.Message;
+import com.zulip.android.models.MessageType;
+import com.zulip.android.models.Person;
+
+import org.json.JSONArray;
+
+public class AsyncEdit extends ZulipAsyncPushTask {
+
+    /**
+     * Initialise an AsyncEdit task to edit a specific message.
+     *
+     * @param zulipActivity The calling Activity
+     * @param msg           The message to edit.
+     */
+    public AsyncEdit(ZulipActivity zulipActivity, Message msg, String messageContent) {
+        super((ZulipApp) zulipActivity.getApplication());
+        this.setProperty("type", msg.getType().toString());
+        if (msg.getType() == MessageType.STREAM_MESSAGE) {
+            this.setProperty("to", msg.getStream().getName());
+        } else {
+            JSONArray arr = new JSONArray();
+            for (Person recipient : msg.getPersonalReplyTo((ZulipApp) zulipActivity.getApplication())) {
+                arr.put(recipient.getEmail());
+            }
+            this.setProperty("to", arr.toString());
```

Comment:
We've recently moved to gson type responses so use GSON type object to json conversion!